src/ray/object_manager/object_manager.cc

```diff
@@ -432,23 +436,36 @@ void ObjectManager::SendObjectChunk(const UniqueID &push_id, const ObjectID &obj
   push_request.set_metadata_size(metadata_size);
   push_request.set_chunk_index(chunk_index);
 
-  // Get data
-  std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> chunk_status =
-      buffer_pool_.GetChunk(object_id, data_size, metadata_size, chunk_index);
-  ObjectBufferPool::ChunkInfo chunk_info = chunk_status.first;
+  if (spilled_object) {
+    // Get data from spilled object.
+    auto optional_chunk = spilled_object->GetChunk(chunk_index);
+    if (!optional_chunk.has_value()) {
+      RAY_LOG(WARNING) << "Attempting to push object " << object_id
+                       << " which is not local. It may have been evicted.";
+      on_complete(Status::IOError("Failed to read spilled object"));
+      return;
+    }
+    push_request.set_data(std::move(optional_chunk.value()));
```

Comment:
Maybe we can do swap here instead? I believe this still has one additional data copy (moved string needs to be copied to the grpc request) whereas Swap will avoid that?